src/models.rs

```diff
@@ -133,12 +133,69 @@ pub struct Page {
     parent: Parent,
 }
 
+impl Identifiable for String {
+    type Type = String;
+
+    fn id(&self) -> &Self::Type {
+        self
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Hash, Clone)]
+#[serde(transparent)]
+pub struct BlockId(String);
+
+impl BlockId {
+    pub fn id(&self) -> &str {
+        &self.0
+    }
+}
+
+impl Identifiable for BlockId {
+    type Type = BlockId;
+
+    fn id(&self) -> &Self::Type {
+        self
+    }
+}
+
+impl Display for BlockId {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
 #[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
-pub struct Block {}
+#[serde(rename_all = "snake_case")]
+pub enum BlockType {
+    Paragraph,
+    Heading1,
+    Heading2,
+    Heading3,
+    BulletedListItem,
+    NumberedListItem,
+    ToDo,
+    Toggle,
+    ChildPage,
+    Unsupported
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
+pub struct Block {
+    id: BlockId,
+    r#type: BlockType,
```

Comment:
Neat, but instead what about 
```suggestion
    #[serde(rename = "type")]
    block_type: BlockType,
```

Reply:
Should block actually be an `enum`?
```rust
pub enum Block {
    Paragraph {
       #[serde(flatten)]
       common: BlockCommon
    }
    ...
```

Reply:
Yeah it needs to be. ill refactor it